### ISSUE TITLE
Why does broadcasting cause `@report_call` errors while `map` does not?

### ISSUE THREAD

**Issue by ehehela** [aviatesk/JET.jl#669]
When I was broadcasting a function over vector of data of custom type, `@report_call` prompts an error stating that no matching method was found. The code is shown below. My question is, why does this error occur? However, when I use `map` function instead of broadcasting, there is no such error. Could this be a false positive?

<details>
  <summary>MWE, Julia version: 1.11.1</summary>

```
using JET, BenchmarkTools

struct Point{dim,T}
    coord::NTuple{dim,T}
end
getcoordinate(n) = n.coord

pts = Point.(rand(NTuple{2,Float64}, 10))

fun(p) = getcoordinate.(p)
coords = fun(pts)
@report_opt fun(pts)    # No errors detected
@report_call fun(pts)   # ═════ 2 possible errors found ═════
@btime fun($pts)    # 17.417 ns (2 allocation: 224 bytes)

fun2(p) = map(getcoordinate, p)
coords = fun2(pts)
@report_opt fun2(pts)    # No errors detected
@report_call fun2(pts)   # No errors detected
@btime fun2($pts)   # 18.437 ns (2 allocation: 224 bytes)
```

</details>

**Comment by aviatesk** [comment]
`map` and broadcast return exactly the same result in this case, but their implementations are different, so it’s not surprising that JET gives different analysis results. However, this report from `@report_call` seems to be a false positive, so I’ll fix it.

**Comment by ehehela** [comment]
Thanks for your selfless contributions and tireless work. This issue can be closed if nothing else.

**Comment by aviatesk** [comment]
This should be fixed in v0.9.12
